packages/gatsby-plugin-google-analytics/src/gatsby-browser.js

```diff
@@ -1,6 +1,20 @@
-exports.onRouteUpdate = function({ location }) {
-  // Don't track while developing.
-  if (process.env.NODE_ENV === `production` && typeof ga === `function`) {
+exports.onRouteUpdate = function({ location }, pluginOptions) {
+  let { enabled } = pluginOptions
+  let isEnabled
+  if (typeof enabled === `boolean`) {
+    isEnabled = enabled
+  } else if (enabled && enabled !== `auto`) {
+    // If user specifies any value that is truthy but NOT the string `auto` it's invalid
+    throw new Error(
+      `[gatsby-plugin-google-analytics] Valid options for 'enabled' flag: true, false, or 'auto'`
+    )
+  } else {
+    // Default value, if `enabled` option is not specified
+    isEnabled = process.env.NODE_ENV === `production`
+  }
```

Comment:
determining if it's disabled or enabled can be done in build time (and also not be duplicated in `gatsby-ssr`). One way to do it, is replicate setup from https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-layout/src/gatsby-node.js:
 - validate and "compute" `isEnabled` flag in `onPreInit`
 - pass computed `isEnabled` to `DefinePlugin` in `onCreateWebpackConfig` (for example as `GATSBY_GOOGLE_ANALYICS_ENABLED`
 - use it in both gatsby-ssr / gatsby-browser: ``if (GATSBY_GOOGLE_ANALYICS_ENABLED && typeof ga === `function`) {``